src/components/article-page.js

```diff
@@ -241,6 +265,30 @@ export default class Article extends PureComponent {
     }
   }
 
+  /**
+   * If users scroll up, and the scrolling distance is more than a certain distance as well, show mobile aside.
+   * Otherwise, if users scroll down, hide the mobile aside.
+   */
+  _toggleMobileAside() {
+    if (this.mobileAsideRef) {
+      const mAside = this.mobileAsideRef
+      const currentTopY = window.scrollY
+
+      // Calculate scrolling distance to determine whether to display aside
+      const lastY = this.scrollPosition.y
+      const distance = currentTopY - lastY
+      if (distance > 30) {
+        this.scrollPosition.y = currentTopY
+        mAside.current.toShow = false
```

Comment:
Will modifying the property of component instance make it re-render?

Reply:
`MobileAside.toShow` is a setter, which would execute `setState({toShow: toShow)` and re-render.